File: 56mergeIntervals/mergeIntervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        result = []
        for interval in sorted(intervals, key = lambda _:_[0]):
            if result and interval[0] <= result[-1][1]: result[-1][1] = max(result[-1][1], interval[1])
            else: result += interval,
        return result

File: 56mergeIntervals/mergeIntervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        std::vector<std::vector<int>> result;
        std::sort(std::begin(intervals), std::end(intervals), [](auto const a, auto const b){return a.front() < b.front();});
        for (auto const& interval: intervals)
        {
            if (!std::empty(result) && interval.front() <= result.back().back()) result.back().back() = std::max(result.back().back(), interval.back());
            else result.emplace_back(interval);
        }
        return result;
    }
};
